### ISSUE TITLE
Score estimate adds captures to Black's score in Chinese rules

### ISSUE THREAD

**Issue by teapoweredrobot** [acristescu/OnlineGo#108]
It seems that when a game is using Chinese rules, the app score estimator adds Black's captures to the score. White's score seems correct.

Can be reproduced by looking at any completed to scoring game under Chinese rules, going back one step, estimating score on the final board position and comparing that score to the actual game score.

White's score seems correct.

**Comment by teapoweredrobot** [comment]
I think this also happens for AGA rules. I've not tested NZ but I guess all area score estimates are affected.

**Comment by acristescu** [comment]
Yes, I can confirm that's how I implemented it, it's mostly Japanese rules. I have to admit that even today the vagaries of the different rule sets confuse me to no end. I would love for somebody more knowledgeable than me to have a crack at this.

**Comment by teapoweredrobot** [comment]
I'm no programmer but I'm happy to have a go at explaining the differences. I think the difficulties would only really arise for unusual situations during play.

For scoring (assuming you are letting the players mark dead stones/groups themselves with no AI or dumb attempts at automatically marking things) then area scoring is literally just the area of stones plus territory on the board (plus Komi) with no regard for prisoners. For handicap games you deduct 1 from black for each handicap stones (I guess no deduction for 1 stone (no Komi) games and then 2 for 2 stones games etc. But would need to check that)

That works for Chinese, NZ and AGA rule sets.

Territory scoring I guess you have correct (territory plus prisoners plus Komi, no adjustment for handicap stones) and works for Japanese rules 

With AGA you can also use territory scoring but in that case you must enforce that white plays last so that potentially there must be three passes to end the game and each pass gives one prisoner to the opponent. So OGS just uses area scoring for AGA rules.

**Comment by acristescu** [comment]
Well I made an attempt at improving the current scoring estimation by
- scoring stones but not prisoners in Chinese and 
- scoring prisoners but not stones for Japanese

I made no attempt to score passes or handicaps as I have no experience with that. The score estimation for a Japanese and a Chinese game I have now agrees with the server in simple examples. Not going to try and tackle seki though...

I'll try to look at the more exotic rulesets in the next few days.
